README.md

```diff
@@ -48,86 +48,109 @@ of your access details to environment variables. To use the `clouds.yaml` file,
 that allow your Go code to access the API - such as the base URL and token ID.
 
-Once we have a base Provider, we inject it as a dependency into each OpenStack
-service. For example, in order to work with the Compute API, we need a Compute
-service client. This can be created like so:
+One single Provider client can be used to build as many Service clients as needed.
 
-```go
-client, err := clientconfig.NewServiceClient("compute", opts)
-```
-
-**Without `gophercloud/utils`**
-
-> *Note*
-> gophercloud doesn't provide support for `clouds.yaml` file so you need to
-> implement this functionality yourself if you don't wish to use
-> `gophercloud/utils`.
-
-You can also generate a "Provider" struct without using the `clientconfig`
-package from `gophercloud/utils`. To do this, you can either pass in your
-credentials explicitly or tell Gophercloud to use environment variables:
+**With `clouds.yaml`**
 
 ```go
+package main
+
 import (
-	"github.com/gophercloud/gophercloud"
-	"github.com/gophercloud/gophercloud/openstack"
+	"context"
+
+	"github.com/gophercloud/gophercloud/v2/openstack"
+	"github.com/gophercloud/gophercloud/v2/openstack/config"
+	"github.com/gophercloud/gophercloud/v2/openstack/config/clouds"
 )
 
-// Option 1: Pass in the values yourself
-opts := gophercloud.AuthOptions{
-  IdentityEndpoint: "https://openstack.example.com:5000/v2.0",
-  Username: "{username}",
-  Password: "{password}",
+func main() {
+	ctx := context.Background()
+
+	// Fetch coordinates from a `cloud.yaml` in the current directory, or
+	// in the well-known config directories (different for each operating
+	// system).
+	authOptions, endpointOptions, tlsConfig, err := clouds.Parse()
+	if err != nil {
+		panic(err)
+	}
+
+	// Call Keystone to get an authentication token, and use it to
+	// construct a ProviderClient. All functions hitting the OpenStack API
+	// accept a `context.Context` to enable tracing and cancellation.
+	providerClient, err := config.NewProviderClient(ctx, authOptions, config.WithTLSConfig(tlsConfig))
+	if err != nil {
+		panic(err)
+	}
+
+	// Use the ProviderClient and the endpoint options fetched from
+	// `clouds.yaml` to build a service client: a compute client in this
+	// case. Note that the contructor does not accept a `context.Context`:
+	// no further call to the OpenStack API is needed at this stage.
+	computeClient, err := openstack.NewComputeV2(providerClient, endpointOptions)
+	if err != nil {
+		panic(err)
+	}
+
+	// use the computeClient
 }
-
-// Option 2: Use a utility function to retrieve all your environment variables
-opts, err := openstack.AuthOptionsFromEnv()
 ```
 
-Once you have the `opts` variable, you can pass it in and get back a
-`ProviderClient` struct:
+**With environment variables (openrc)**
+
+Gophercloud can parse the environment variables set by running `source openrc`:
 
 ```go
-provider, err := openstack.AuthenticatedClient(context.TODO(), opts)
+opts, err := openstack.AuthOptionsFromEnv()
```

Comment:
Could we wrap this in a main function and add the relevant imports, so that a user can diff things side-by-side?